pkg/node/manager/manager.go

```diff
@@ -115,6 +118,37 @@ func NewNodeManager(logger logr.Logger, resourceManager resource.ResourceManager
 	return manager, worker.StartWorkerPool(manager.performAsyncOperation)
 }
 
+func (m *manager) CheckNodeForLeakedENIs(nodeName string) {
+	managedNode, found := m.GetNode(nodeName)
+	if !found {
+		m.Log.Info("Node manager couldn't find the node for reconciliation cleanup", "NodeName", nodeName)
+		return
+	}
+
+	// Only start a goroutine when need to
+	if time.Now().After(managedNode.GetNextReconciliationTime()) {
+		go func() {
+			if resourceProvider, found := m.resourceManager.GetResourceProvider(config.ResourceNamePodENI); found {
+				foundLeakedENI := resourceProvider.ReconcileNode(nodeName)
+				if foundLeakedENI {
+					managedNode.SetReconciliationInterval(node.NodeInitialCleanupInterval)
+				} else {
+					interval := wait.Jitter(managedNode.GetReconciliationInterval(), 5)
+					if interval > node.MaxNodeReconciliationInterval {
+						interval = node.MaxNodeReconciliationInterval
+					}
+					managedNode.SetReconciliationInterval(interval)
+				}
+				managedNode.SetNextReconciliationTime(time.Now().Add(managedNode.GetReconciliationInterval()))
+				m.Log.Info("reconciled cleanup node for leaking branch interfaces", "NodeName", nodeName, "NextInterval", managedNode.GetReconciliationInterval(), "NextReconciliationTime", managedNode.GetNextReconciliationTime())
+			} else {
+				// no SGP provider enabled
```

Comment:
In this case, should we set next reconciliation time to never? Otherwise, the `if` statement starting the routine above will always return true

Reply:
Do you mean if the node is not using SGP feature? Yes, in this case, every time when a node event triggers a check a goroutine is created but immediately returns. I think there shouldn't be impact on performance since they are light and very short lifespan.

Reply:
Yeah, discussed offline and this is fine for now. We can improve in the future with finalizers if needed